blockdag/reachability.go

```diff
@@ -187,33 +198,78 @@ type reachabilityTreeNode struct {
 	// interval is the index interval containing all intervals of
 	// blocks in this node's subtree
 	interval *reachabilityInterval
-
-	// remainingInterval is the not-yet allocated interval (within
-	// this node's interval) awaiting new children
-	remainingInterval *reachabilityInterval
 }
 
 func newReachabilityTreeNode(blockNode *blockNode) *reachabilityTreeNode {
 	// Please see the comment above reachabilityTreeNode to understand why
 	// we use these initial values.
 	interval := newReachabilityInterval(1, math.MaxUint64-1)
-	// We subtract 1 from the end of the remaining interval to prevent the node from allocating
+	return &reachabilityTreeNode{blockNode: blockNode, interval: interval}
+}
+
+func (rtn *reachabilityTreeNode) childIntervalAllocationRange() *reachabilityInterval {
```

Comment:
Maybe child**ren**IntervalAllocationRange ?

Reply:
That's a bit ungrammatical. I renamed it to intervalRangeForChildAllocation to make it more clear.